### PR TITLE
fix besu retesteth PoW mining

### PR DESCRIPTION
**Pull request by garyschulte**
Signed-off-by: Gary Schulte <gary@Garys-MacBook-Pro.local>

## PR description
it appears that [#1882] inadvertently broke retesteth no-op PoW mining.  This is a quickie PR to get it moving again.

## Changelog

- [X] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/PoWSolver.java [hunk 1]
```diff
@@ -127,11 +127,19 @@ private void findValidNonce() {
   }
 
   private Optional<PoWSolution> testNonce(final PoWSolverInputs inputs, final long nonce) {
-    PoWSolution solution =
-        poWHasher.hash(nonce, inputs.getBlockNumber(), epochCalculator, inputs.getPrePowHash());
-    final UInt256 x = UInt256.fromBytes(solution.getSolution());
```
[the remark this answers is not in the document]
Comment (garyschulte) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/PoWSolver.java:132]: Yeah, the NO_WORK_HASHER from RetestethContext was returning null from`.hash(...)`

[the remark this answers is not in the document]
Comment (RatanRSur) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/PoWSolver.java:132]: If so, do we still need to use `Optional`? Or is `hash` now defined at all locations?

[the remark this answers is not in the document]
Comment (garyschulte) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/PoWSolver.java:132]: well, the return type is Optional<PoWSolution> so it just seems natural to plumb it through

[the remark this answers is not in the document]
Comment (RatanRSur) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/PoWSolver.java:132]: Ah yes, ok, in that case, I'd suggest
```
  private Optional<PoWSolution> testNonce(final PoWSolverInputs inputs, final long nonce) {
    return Optional.ofNullable(
            poWHasher.hash(nonce, inputs.getBlockNumber(), epochCalculator, inputs.getPrePowHash()))
        .filter(sol -> UInt256.fromBytes(sol.getSolution()).compareTo(inputs.getTarget()) <= 0);
  }
```

**Review by RatanRSur — APPROVED** [review]

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/PoWSolver.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (garyschulte) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/PoWSolver.java:132]: Yeah, that is nice and lean  👍

**Pull request merged**